rust/datafusion/src/physical_plan/functions.rs

```diff
@@ -494,6 +532,27 @@ pub fn return_type(
                 ));
             }
         }),
+        BuiltinScalarFunction::SplitPart => Ok(match arg_types[0] {
+            DataType::LargeUtf8 => DataType::LargeUtf8,
+            DataType::Utf8 => DataType::Utf8,
+            _ => {
+                // this error is internal as `data_types` should have captured this.
+                return Err(DataFusionError::Internal(
+                    "The split_part function can only accept strings.".to_string(),
+                ));
+            }
+        }),
+        BuiltinScalarFunction::StartsWith => Ok(DataType::Boolean),
```

Comment:
I wonder if there is a reason not to check the arg types of `starts_with` given we are checking all the others?

Reply:
Because this is just the return type and will always be Boolean whereas something like `InitCap` will return either `Utf8`  or `LargeUtf8` depending on input 😁